src/js/add_input_icon.js

```diff
@@ -240,14 +238,44 @@ async function addRelayIconToInput(emailInput) {
       sendInPageEvent("click", "input-menu-manage-all-aliases-btn");
     });
 
+    //Create "Get unlimited aliases" link
+    const { fxaSubscriptionsUrl } = await browser.storage.local.get("fxaSubscriptionsUrl");
+    const { premiumProdId } = await browser.storage.local.get("premiumProdId");
+    const { premiumPriceId } = await browser.storage.local.get("premiumPriceId");
+    getUnlimitedAliasesBtn.href = `${fxaSubscriptionsUrl}/products/${premiumProdId}?plan=${premiumPriceId}`;
+
     // Restrict tabbing to relay menu elements
     restrictOrRestorePageTabbing(-1);
 
     // Append menu elements to the menu
-    [generateAliasBtn, remainingAliasesSpan, relayMenuDashboardLink].forEach(el => {
+    [remainingAliasesSpan, getUnlimitedAliasesBtn, generateAliasBtn, relayMenuDashboardLink].forEach(el => {
       relayInPageMenu.appendChild(el);
     });
 
+    //Show get unlimited aliases btn
+    if (!premium) {
+      if (maxNumAliasesReached) {
+        generateAliasBtn.remove();
+        sendInPageEvent("viewed-menu", "input-menu-max-aliases-message");
+        remainingAliasesSpan.textContent = browser.i18n.getMessage("pageFillRelayAddressLimit", [numAliasesRemaining, maxNumAliases]);
+      }
+
+      else {
+        getUnlimitedAliasesBtn.remove();
+      }
+    }
+
+    else {
+      getUnlimitedAliasesBtn.remove();
+    }
+
+    const premiumEnabled = await browser.storage.local.get("premiumEnabled");
+    const premiumEnabledString = premiumEnabled.premiumEnabled;
+
+    if(premiumEnabledString != "True") {
+      getUnlimitedAliasesBtn.remove();
+    }
```

Comment:
I feel like you could add this if statement to the above main check and chain it with some or `||` logic.